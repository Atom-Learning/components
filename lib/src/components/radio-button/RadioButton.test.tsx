import { render, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import React from 'react'

import { RadioButton, RadioButtonGroup } from '.'

describe(`Radio component`, () => {
  it('renders a radio button', async () => {
    const { container } = render(
      <RadioButtonGroup>
        <RadioButton value="value" aria-label="indicator" />
      </RadioButtonGroup>
    )

    await screen.getByRole('radio')

    expect(container).toMatchSnapshot()
  })

  it('renders a radio button - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <label>
        Label
        <RadioButtonGroup>
          <RadioButton value="value" aria-label="indicator" />
        </RadioButtonGroup>
      </label>
    )

    expect(await axe(container)).toHaveNoViolations()
  })

  it('renders a disabled radio button', async () => {
    const { container } = render(
      <RadioButtonGroup>
        <RadioButton value="value" aria-label="disabled indicator" disabled />
      </RadioButtonGroup>
    )

    await screen.getByRole('radio')

    expect(container).toMatchSnapshot()
  })

  it('renders a disabled radio button - has no programmatically detectable a11y issues', async () => {
    const { container } = render(
      <label>
        Label
        <RadioButtonGroup>
          <RadioButton value="value" aria-label="disabled indicator" disabled />
        </RadioButtonGroup>
      </label>
    )

    await screen.getByRole('radio')

    expect(await axe(container)).toHaveNoViolations()
  })
})
