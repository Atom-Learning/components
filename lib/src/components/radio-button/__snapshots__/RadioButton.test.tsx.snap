// Vitest Snapshot v1

exports[`Radio component > renders a disabled radio button 1`] = `
<div>
  <div
    class="c-iSwBqf"
    dir="ltr"
    role="radiogroup"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-checked="false"
      aria-label="disabled indicator"
      class="c-hXfYPB"
      data-disabled=""
      data-radix-collection-item=""
      data-state="unchecked"
      disabled=""
      role="radio"
      tabindex="-1"
      type="button"
      value="value"
    />
  </div>
</div>
`;

exports[`Radio component > renders a radio button 1`] = `
<div>
  <div
    class="c-iSwBqf"
    dir="ltr"
    role="radiogroup"
    style="outline: none;"
    tabindex="0"
  >
    <button
      aria-checked="false"
      aria-label="indicator"
      class="c-hXfYPB"
      data-radix-collection-item=""
      data-state="unchecked"
      role="radio"
      tabindex="-1"
      type="button"
      value="value"
    />
  </div>
</div>
`;
